analyses/infercnv-consensus-cell-type/scripts/01_run-infercnv.R

```diff
@@ -0,0 +1,241 @@
+  ),
+  make_option(
+    opt_str = "--reference_file",
+    type = "character",
+    help = "Path to the normal reference SCE"
+  ),
+  make_option(
+    opt_str = c("--output_dir"),
+    type = "character",
+    help = "Folder to save final infercnv results"
+  ),
+  make_option(
+    opt_str = c("--hmm_model"),
+    type = "character",
+    default = "i3",
+    help = "The HMM model to use with inferCNV, either 'i3' or 'i6'. Default is i3."
+  ),
+  make_option(
+    opt_str = c("--skip_hmm"),
+    action = "store_true",
+    default = FALSE,
+    help = "Use this flag to turn off fitting the HMM."
+  ),
+  # clustering options. context:
+  # https://github.com/broadinstitute/infercnv/wiki/infercnv-tumor-subclusters#tumor-subclustering-by-leiden-clustering-preferred
+  make_option(
+    opt_str = c("--leiden_function"),
+    type = "character",
+    default = "modularity", # overrides their default of CPM
+    help = "Objective function to use during leiden clustering"
+  ),
+  make_option(
+    opt_str = c("--leiden_method"),
+    type = "character",
+    default = "PCA",
+    help = "Data on which clustering should be performed, either PCA or simple (expression data itself) for leiden clustering"
+  ),
+  make_option(
+    opt_str = c("--leiden_resolution"),
+    type = "numeric",
+    default = 1,
+    help = "Resolution value to use with leiden clsutering, which will use modularity by default"
+  ),
+  make_option(
+    opt_str = c("--gene_order_file"),
+    type = "character",
+    default = file.path(project_root, "references", "Homo_sapiens.GRCh38.104.gene_order.txt"),
+    help = "Path to gene order file as tab delimited .txt file with no column headers.
+      Columns are: Ensembl gene id, chr, start, stop."
+  ),
+  make_option(
+    opt_str = "--annotation_file",
+    type = "character",
+    default = file.path(project_root, "scratch", "infercnv_annotation.txt"),
+    help = "Path to annotation file for inferCNV input"
+  ),
+  make_option(
+    opt_str = c("--scratch_dir"),
+    type = "character",
+    default = file.path(project_root, "scratch", "infercnv"),
+    help = "path to scratch directory to store intermediate files from infercnv"
+  ),
+  make_option(
+    opt_str = c("-t", "--threads"),
+    type = "integer",
+    default = 4,
+    help = "Number of multiprocessing threads to use."
+  ),
+  make_option(
+    opt_str = c("--seed"),
+    type = "integer",
+    default = 2025,
+    help = "Random seed to set for reproducibility. Note that inferCNV is only reproducible on a given operating system."
+  )
+)
+
+
+# Parse options ----------------
+opts <- parse_args(OptionParser(option_list = option_list))
+
+
+# check input directories
+stopifnot(
+  "sce_file does not exist" = file.exists(opts$sce_file),
+  "reference_file does not exist" = file.exists(opts$reference_file),
+  "gene_order_file does not exist" = file.exists(opts$gene_order_file),
+  "annotation_file not provided" = !is.null(opts$annotation_file),
```

Comment:
```suggestion
  "annotation_file not provided" = file.exists(opts$annotation_file),
```
Since we set a default for this, we want to just make sure it exists.

Reply:
This was a revision you previously left, but when I tested it I realized doesn't actually work as the code is written - we do have a default for it, but the file itself doesn't exist because it's written out as part of the script itself. But, what I can also check for is that its `dirname` exists?

Reply:
noting i added that additional check here! https://github.com/AlexsLemonade/OpenScPCA-analysis/pull/1095/commits/d4be9c673c1389ab05baae47363e831d9269188b

Reply:
Ah that makes sense! I do think rather than check that the directory exists you could just make the directory to ensure that it exists before writing out.